src/core/utils/otns.hpp

```diff
@@ -36,22 +36,24 @@
 
 #include "openthread-core-config.h"
 
-#if (OPENTHREAD_MTD || OPENTHREAD_FTD) && OPENTHREAD_CONFIG_OTNS_ENABLE
+#if OPENTHREAD_CONFIG_OTNS_ENABLE
 
 #include <openthread/thread.h>
 #include <openthread/thread_ftd.h>
 #include <openthread/platform/otns.h>
-
-#include "coap/coap_message.hpp"
 #include "common/locator.hpp"
 #include "common/non_copyable.hpp"
-#include "common/notifier.hpp"
 #include "common/string.hpp"
 #include "mac/mac_frame.hpp"
 #include "mac/mac_types.hpp"
+
+#if OPENTHREAD_MTD || OPENTHREAD_FTD
+#include "coap/coap_message.hpp"
+#include "common/notifier.hpp"
 #include "net/ip6_address.hpp"
 #include "thread/neighbor.hpp"
 #include "thread/neighbor_table.hpp"
+#endif // OPENTHREAD_MTD || OPENTHREAD_FTD
```

Comment:
We do not need to add the `// OPENTHREAD_MTD || OPENTHREAD_FTD` comment if the corresponding `#if` block contains fewer than 20 lines.

The idea behind these comments is to make it easier to match an `#endif` with its corresponding `#if` when they are far apart. However, if they are already close, the comment is not needed and can make the code less readable by adding visual clutter.

We have opted for the "20-line rule" in the Openthread [style guide](https://github.com/openthread/openthread/blob/c55098af5e62f4f388d9c4e7f5bfbaa6a6ac687b/STYLE_GUIDE.md):